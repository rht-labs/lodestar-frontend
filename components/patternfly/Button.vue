<template>
  <button class="pf-c-button pf-m-primary">
    {{ text }}
  </button>
</template>

<script>
export default {
  props: {
    text: {
      type: String,
      default: 'Button!'
    }
  }
}
</script>
