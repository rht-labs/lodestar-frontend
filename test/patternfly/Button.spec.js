import { mount } from '@vue/test-utils'
import Button from '@/components/patternfly/Button.vue'

describe('Button', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Button)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = mount(Button)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
