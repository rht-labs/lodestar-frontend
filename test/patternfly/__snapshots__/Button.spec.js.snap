// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders properly 1`] = `
"<button class=\\"pf-c-button pf-m-primary\\">
  Button!
</button>"
`;
